Index: app/src/main/res/xml/preferences.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/res/xml/preferences.xml	(date 1476950987000)
+++ app/src/main/res/xml/preferences.xml	(revision )
@@ -47,8 +47,18 @@
 	            android:key="showNextProgramPref"
 	            android:summary="@string/pref_show_next_program_sum"
 	            android:title="@string/pref_show_next_program" />
-	        <CheckBoxPreference
-	            android:defaultValue="true"
+            <CheckBoxPreference
+                android:defaultValue="true"
+                android:key="showProgramProgressbarPref"
+                android:summary="@string/pref_program_progressbar_sum"
+                android:title="@string/pref_program_progressbar" />
+            <CheckBoxPreference
+                android:defaultValue="true"
+                android:key="showChannelNamePref"
+                android:summary="@string/pref_show_channel_name_sum"
+                android:title="@string/pref_show_channel_name" />
+	        <CheckBoxPreference
+	            android:defaultValue="true"
 	            android:key="playWhenChannelIconSelectedPref"
 	            android:summary="@string/pref_play_when_channel_icon_selected_sum"
 	            android:title="@string/pref_play_when_channel_icon_selected" />
\ No newline at end of file
Index: app/src/main/res/values/strings.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/res/values/strings.xml	(date 1476950987000)
+++ app/src/main/res/values/strings.xml	(revision )
@@ -370,4 +370,9 @@
 
     <string name="parent_directory">Parent directory</string>
 
+    <string name="pref_program_progressbar_sum">The elapsed time of a program will be shown as a horizonal progressbar.</string>
+    <string name="pref_program_progressbar">Show program progress</string>
+    <string name="pref_show_channel_name_sum">The name of the channel will be shown additionally to the icon.</string>
+    <string name="pref_show_channel_name">Show channel name</string>
+
 </resources>
Index: app/src/main/java/org/tvheadend/tvhclient/Utils.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/java/org/tvheadend/tvhclient/Utils.java	(date 1476950987000)
+++ app/src/main/java/org/tvheadend/tvhclient/Utils.java	(revision )
@@ -751,11 +751,11 @@
     /**
      * Shows the progress as a progress bar.
      *
-     * @param progress Progress bar widget
+     * @param progress Progress widget
      * @param start    Start time
      * @param stop     Stop time
      */
-    public static void setProgress(ProgressBar progress, final Date start, final Date stop) {
+    public static void setProgress(View progress, final Date start, final Date stop) {
         if (progress == null || start == null || stop == null) {
             return;
         }
@@ -768,8 +768,13 @@
         if (durationTime > 0) {
             percent = elapsedTime / durationTime;
         }
-        progress.setProgress((int) Math.floor(percent * 100));
-        progress.setVisibility(View.VISIBLE);
+
+        final int value = (int) Math.floor(percent * 100);
+        if (progress instanceof ProgressBar) {
+            ((ProgressBar) progress).setProgress(value);
+        } else if (progress instanceof TextView) {
+            ((TextView) progress).setText(String.valueOf(value) + "%");
+        }
     }
 
     /**
Index: app/src/main/res/layout/channel_list_widget_item.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/res/layout/channel_list_widget_item.xml	(date 1476950987000)
+++ app/src/main/res/layout/channel_list_widget_item.xml	(revision )
@@ -98,7 +98,7 @@
                 style="?attr/ContentFont"
                 android:layout_width="0dip"
                 android:layout_height="wrap_content"
-                android:layout_weight="0.7"
+                android:layout_weight="0.65"
                 android:paddingEnd="16dp"
                 android:paddingLeft="0dp"
                 android:paddingRight="16dp"
@@ -111,10 +111,20 @@
                 style="?attr/ContentFont"
                 android:layout_width="0dip"
                 android:layout_height="wrap_content"
-                android:layout_weight="0.3"
+                android:layout_weight="0.25"
                 android:gravity="end"
                 android:singleLine="true"
                 android:text="@string/duration" />
+
+            <TextView
+                android:id="@+id/progress"
+                style="?attr/ContentFont"
+                android:layout_width="0dip"
+                android:layout_height="wrap_content"
+                android:layout_weight="0.1"
+                android:gravity="end"
+                android:singleLine="true"
+                android:text="@null" />
         </LinearLayout>
 
         <ProgressBar
\ No newline at end of file
Index: app/src/main/java/org/tvheadend/tvhclient/adapter/ChannelListAdapter.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/java/org/tvheadend/tvhclient/adapter/ChannelListAdapter.java	(date 1476950987000)
+++ app/src/main/java/org/tvheadend/tvhclient/adapter/ChannelListAdapter.java	(revision )
@@ -93,6 +93,7 @@
         public TextView channel;
         public TextView time;
         public TextView duration;
+        public TextView progress;
         public ProgressBar progressbar;
         public ImageView state;
         public TextView genre;
@@ -117,6 +118,7 @@
             holder.title = (TextView) view.findViewById(R.id.title);
             holder.nextTitle = (TextView) view.findViewById(R.id.next_title);
             holder.channel = (TextView) view.findViewById(R.id.channel);
+            holder.progress = (TextView) view.findViewById(R.id.progress);
             holder.progressbar = (ProgressBar) view.findViewById(R.id.progressbar);
             holder.time = (TextView) view.findViewById(R.id.time);
             holder.duration = (TextView) view.findViewById(R.id.duration);
@@ -151,8 +153,12 @@
             // Set the initial values
             if (holder.progressbar != null) {
                 holder.progressbar.setProgress(0);
-                holder.progressbar.setVisibility(prefs.getBoolean("showProgramProgressPref", true) ? View.VISIBLE : View.GONE);
+                holder.progressbar.setVisibility(prefs.getBoolean("showProgramProgressbarPref", true) ? View.VISIBLE : View.GONE);
             }
+            if (holder.progress != null) {
+                holder.progress.setText("0%");
+                holder.progress.setVisibility(prefs.getBoolean("showProgramProgressbarPref", true) ? View.GONE : View.VISIBLE);
+            }
             if (holder.channel != null) {
                 holder.channel.setText(c.name);
                 holder.channel.setVisibility(prefs.getBoolean("showChannelNamePref", true) ? View.VISIBLE : View.GONE);
@@ -248,6 +254,7 @@
                 Utils.setTime(holder.time, p.start, p.stop);
                 Utils.setDuration(holder.duration, p.start, p.stop);
                 Utils.setProgress(holder.progressbar, p.start, p.stop);
+                Utils.setProgress(holder.progress, p.start, p.stop);
 
                 if (holder.nextTitle != null && np != null) {
                     holder.nextTitle.setVisibility(prefs.getBoolean("showNextProgramPref", true) ? View.VISIBLE : View.GONE);
@@ -262,6 +269,9 @@
                 }
                 if (holder.progressbar != null) {
                     holder.progressbar.setVisibility(View.GONE);
+                }
+                if (holder.progress != null) {
+                    holder.progress.setVisibility(View.GONE);
                 }
                 if (holder.time != null) {
                     holder.time.setVisibility(View.GONE);
